Userland/Libraries/LibTLS/TLSv12.cpp

```diff
@@ -499,18 +499,21 @@ DefaultRootCACertificates::DefaultRootCACertificates()
         return;
     }
     auto data = data_result.release_value();
-    reload_certificates(data);
-}
 
-void DefaultRootCACertificates::reload_certificates(ByteBuffer& data)
-{
-    auto decode_result = Crypto::decode_pems(data);
-    if (decode_result.is_error()) {
-        dbgln("Failed to load CA Certificates: {}", decode_result.error());
+    auto reload_result = reload_certificates(data);
+    if (reload_result.is_error()) {
+        dbgln("Failed to load CA Certificates: {}", reload_result.error());
         return;
     }
-    m_ca_certificates.clear();
-    auto certs = decode_result.release_value();
+
+    m_ca_certificates = reload_result.release_value();
+}
+
+ErrorOr<Vector<Certificate>> DefaultRootCACertificates::reload_certificates(ByteBuffer& data)
```

Comment:
should be static. probably also renamed? (e.g. `load_certificates` or `parse_certificates`? in fact it is bound for root ca certificates so putting that explicitly in the name would be a merit!). not sure how much it belongs to that class—it just parsing/loading with additional validation…

note that during loading ca certificates into a trust store it need not to be self-issued nor self-signed certificate — definitely it should be a ca certificate though. (end-entity certificates could be handled separately.)

Reply:
I didn't want to change it too much in a pretty much unrelated PR, so I changed the absolute minimum to deduplicate code. How the function is setup should definitely be reworked in another PR.